Lib/sysconfig.py

```diff
@@ -545,15 +545,32 @@ def _init_non_posix(vars):
     """Initialize the module as appropriate for NT"""
     # set basic install directories
     import _imp
+    import re
     vars['LIBDEST'] = get_path('stdlib')
     vars['BINLIBDEST'] = get_path('platstdlib')
     vars['INCLUDEPY'] = get_path('include')
-    try:
+
+    extension_suffixes = _imp.extension_suffixes()
+    if len(extension_suffixes) >= 1:
         # GH-99201: _imp.extension_suffixes may be empty when
         # HAVE_DYNAMIC_LOADING is not set. In this case, don't set EXT_SUFFIX.
-        vars['EXT_SUFFIX'] = _imp.extension_suffixes()[0]
-    except IndexError:
-        pass
+        # e.g., "_d.cp313-win_amd64.pyd"
+        vars['EXT_SUFFIX'] = extension_suffixes[0]
+
+        # e.g., "cp313-win_amd64"
+        _, soabi, _ = vars['EXT_SUFFIX'].split('.')
+        vars['SOABI'] = soabi
+
+        # e.g., check for a "t" (for "threading") in SOABI
+        if re.match(r'cp\d+t', soabi):
+            vars['Py_NOGIL'] = 1
+
+    vars['LIBPL'] = _safe_realpath(os.path.join(get_config_var('installed_base'), 'libs'))
+    if hasattr(sys, 'dllhandle'):
```

Comment:
I think that's correct, but would like if someone else confirmed it. @zooba?

Reply:
Until the `NO_ENABLE_SHARED` code is removed, I'd rather not assume it (the code in `ctypes` should be fixed too). The standard library should be usable with other runtimes as well, and while `sysconfig` is pretty closely tied to specifically CPython, the less others have to modify the better.

Reply:
Should we open an issue to discuss the fate of `NO_ENABLE_SHARED`? So that we, at least, clarify it.

Reply:
Opened GH-110234, but after taking a deeper look into the issue, I think support probably won't be removed as it's low maintenance. The `sys.dllhandle` documentation just needs to note that it is only available on dynamic library Windows builds, because right now it reads as it is always available on Windows, explaining why people are writing code with that assumption.

Good find @colesbury!